client/my-sites/checkout/checkout/wechat-payment-box.jsx

```diff
@@ -0,0 +1,258 @@
+		errorNotice: PropTypes.func,
+		infoNotice: PropTypes.func,
+	};
+
+	state = {
+		name: '',
+		errorMessage: '',
+	};
+
+	componentWillUnmount() {
+		this.props.resetWeChatOrder();
+	}
+
+	handleSubmit = event => {
+		event.preventDefault();
+		const validation = validatePaymentDetails( { name: this.state.name }, 'wechat' );
+		if ( ! isEmpty( validation.errors ) ) {
+			return this.setState( { errorMessage: validation.errors[ 0 ] } );
+		}
+
+		this.props.fetchWeChatOrder( this.state.name );
+	};
+
+	componentDidUpdate( prevProps ) {
+		const {
+			resetWeChatOrder,
+			cart,
+			weChatOrderData: { redirectUrl },
+			isWeChatOrderDataError,
+			errorNotice,
+			infoNotice,
+		} = this.props;
+
+		if ( isWeChatOrderDataError ) {
+			errorNotice( translate( "We've encountered a problem. Please try again later." ) );
+		}
+
+		// The Wechat payment type should only redirect when on mobile as redirect urls
+		// are Wechat Pay mobile application urls: e.g. weixin://wxpay/bizpayurl?pr=RaXzhu4
+		if ( userAgent.isMobile && redirectUrl ) {
+			infoNotice(
+				translate( 'We are now redirecting you to the WeChat Pay mobile app to finalize payment.' )
+			);
+			return location.assign( redirectUrl );
+		}
+
+		// Reset transaction if cart is modified
+		if (
+			prevProps.cart.total_cost !== cart.total_cost ||
+			prevProps.cart.products.length !== cart.products.length
+		) {
+			resetWeChatOrder();
+		}
+
+		if ( redirectUrl && ! userAgent.isMobile ) {
+			// Display on desktop
+			infoNotice(
+				translate( 'Please scan the WeChat Payment barcode.', {
+					comment: 'Instruction to scan the on screen barcode.',
+				} )
+			);
+		}
+	}
+
+	render() {
+		const {
+			weChatOrderData: { redirectUrl, orderId },
+			selectedSite,
+			presaleChatAvailable,
+			cart,
+
+			paymentType,
+			children,
+			isRequestingWeChatOrderData,
+		} = this.props;
+
+		// Only show if chat is available and we have a business plan in the cart.
+		const showPaymentChatButton =
+			presaleChatAvailable && some( cart.products, isWpComBusinessPlan );
+
+		// Wechat qr codes get set on desktop instead of redirecting
+		if ( redirectUrl && ! userAgent.isMobile ) {
+			return (
+				<WeChatPaymentQRcode
+					orderId={ orderId }
+					cart={ cart }
+					redirectUrl={ redirectUrl }
+					slug={ selectedSite.slug }
+				/>
+			);
+		}
+
+		const formClasses = classNames( {
+			'is-loading': isRequestingWeChatOrderData,
+		} );
+
+		return (
+			<React.Fragment>
+				<form onSubmit={ this.handleSubmit } className={ formClasses }>
+					<div className="checkout__payment-box-sections">
+						<div className="checkout__payment-box-section">
+							<Input
+								additionalClasses="checkout__checkout-field"
+								label={ translate( 'Your Name' ) }
+								isError={ ! isEmpty( this.state.errorMessage ) }
+								errorMessage={ this.state.errorMessage }
+								name="name"
+								onBlur={ event => this.setState( { name: event.target.value } ) }
```

Comment:
I think we tend to avoid inline functions in `render()`. JS allocates a new func instance every time render is called. So we can create a method instead:
```
onChange = event => this.setState( { name: event.target.value } );
``